app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,74 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = Recipe.where(public: false, author_id: current_user.id)
+    respond_to do |format|
+      format.html
+      format.json do
+        if user_signed_in?
+          render json: @recipes
+        else
+          render html: 'you are not allowed to see this content.'
+        end
+      end
+    end
+  end
+
+  def public_recipes
```

Comment:
All the public recipes should have the information of the **total price** and **total food items** associated with them which is missing in your case. You can refer to [this video](https://www.loom.com/share/4e90dd1f3531432689584a4b4d9e5c53) where I demonstrated the problem in detail. Kindly fix this issue as this is a part of the core requirements for this project. Below is the screenshot where you can see how to implement the features. ⬇️
![image](https://user-images.githubusercontent.com/33268927/209438000-af571bb2-b03f-4182-8eb2-000e93bd9062.png)